packages/beacon-state-transition/src/fast/stateTransition.ts

```diff
@@ -26,7 +26,10 @@ export function fastStateTransition(
   // process slots (including those with no blocks) since block
   switch (preFork) {
     case "phase0":
-      processSlots(postState as CachedBeaconState<phase0.BeaconState>, block.slot);
+      if (postState.slot < block.slot) {
```

Comment:
This condition can be added to `processSlots` by tweaking the assertion there. eg: `assert.lte(state.slot, block.slot)`